pallets/identity/src/lib.rs

```diff
@@ -114,12 +123,26 @@ pub mod pallet {
         #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
         /// Add a new identity trait to identity_id with key/value.
         pub fn add_identity_trait(origin: OriginFor<T>, identity_id: u32, key: Vec<u8>, value: Vec<u8>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            ensure!(Self::is_identity_owned_by_sender(&who, &identity_id), Error::<T>::IdentityNotOwned);
+
+            <IdentityTraitList<T>>::insert(identity_id, key, value);
```

Comment:
going to replace this with `try_mutate` -- do we want to change the naming of this method to `upsert`?

Reply:
Yeah, I think `try_mutate` makes sense here - what's your reasoning for changing the naming?

Reply:
Do you want them to be able to edit a field on an identity? If so, we don't have a method for editing. We could add a `edit_identity_trait` or, we could actually use this method, especially if we replace `insert` with `try_mutate` -- it would have the same method signature and work both ways.

Reply:
Ahh, I get what you're saying. I think we should split that into two extrinsics, one each for add/update. Makes it a bit more explicit and avoids giving any one function too many jobs.